pulsar-broker/src/main/java/org/apache/pulsar/broker/web/AuthenticationFilter.java

```diff
@@ -60,8 +63,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             if (LOG.isDebugEnabled()) {
                 LOG.debug("[{}] Authenticated HTTP request with role {}", request.getRemoteAddr(), role);
             }
-        } catch (AuthenticationException e) {
+        } catch (PulsarHttpAuthenticationException e) {
             HttpServletResponse httpResponse = (HttpServletResponse) response;
+            if (isNotBlank(e.getRealmInformation())) {
+                ((HttpServletResponse) response).setHeader("WWW-Authenticate", e.getRealmInformation());
```

Comment:
As I wrote on the previous comment, there is another standard header that is designed for returning information on authentication success, and authentication methods are also allowed to use other non-standard headers. Requiring auth providers to embed the header information into an exception and assuming the header name is "WWW-Authenticate" would be a limitation in the future.

Honestly speaking, I think using an exception to indicate simple authentication failure is a bad design, at least in Java, because of its cost. Authentication failures happen occasionally, and it can be caused easily by attackers. We shouldn't use exceptions for such a expectable case.

Reply:
@maskit 
I roughly agree with you.

> using an exception to indicate simple authentication failure is a bad design

I think this should not be fixed in this PR adding authentication plugin.
We should consider good design and make another PR.
That's why, I exclude the code about authentication header from this PR.

Reply:
Alright.